File: C++/Calc.cpp
#include <iostream>
#include <string>
using namespace std;

int calc(char op, string in){
    string i1 = in.substr(0, in.find(op));
    string i2 = in.substr(in.find(op) + 1, in.length());
    switch (op) {
        case '+':
            return stoi(i1) + stoi(i2);

        case '-':
           return stoi(i1) - stoi(i2);

        case '/':
            return stoi(i1) / stoi(i2);
        
        case '*':
            return stoi(i1) * stoi(i2);

        case '%':
            return stoi(i1) % stoi(i2);
    }
    return 0;
}

int main(){
    cout << "Hello from Calc in C++! \n";
    cout << "Type \"exit\" to exit. \n";

    while (true)
    {
        string in = "";
        cout << ">";
        cin >> in;

        if(in.find("+") != 18446744073709551615 && in.find("+") != 0){
            cout << calc('+', in);

        } else if(in.find("-") != 18446744073709551615 && in.find("+") != 0){
            cout << calc('-', in);

        } else if(in.find("/") != 18446744073709551615 && in.find("+") != 0){
            cout << calc('/', in);
  
        } else if(in.find("*") != 18446744073709551615 && in.find("+") != 0){
            cout << calc('*', in);
  
        } else if(in.find("%") != 18446744073709551615 && in.find("+") != 0){
            cout << calc('%', in);

        } else if(in.find("exit") != 18446744073709551615 && in.find("+") != 0){
            return 0;

        } else {
            cout << "No correct placed operator or not supported operator.";
        }

        cout << "\n";
    }
    
    return 0;
}

File: Python/Calc.py
def calc(line, operator):
    ls = line.split(operator)
    
    match operator:
        case "+":
            print(int(ls[0]) + int(ls[1]))
        
        case "-":
            print(int(ls[0]) - int(ls[1]))
        
        case "/":
            print(int(ls[0]) / int(ls[1]))
                
        case "*":
            print(int(ls[0]) * int(ls[1]))
        
        case "%":
            print(int(ls[0]) % int(ls[1]))    
    
print("Hello from Calc in Python!")
print("Type \"exit\" to exit.")

while True:
    line = input(">")   
    
    if line == "":
        print("You need to type sth.")
    else:
        if "exit" in line:
            exit(0)

        if "+" in line:
            calc(line, "+")
            
        elif "-" in line:
            calc(line, "-")

        elif "/" in line:
            calc(line, "/")

        elif "*" in line:
            calc(line, "*")

        elif "%" in line:
            calc(line, "%")
        else:
            print("No operator or not supported operator.")
